src/System.Collections/src/System/Collections/Generic/HashSetEqualityComparer.cs

```diff
@@ -50,13 +87,10 @@ public override bool Equals(object obj)
             {
                 return false;
             }
-            return (_comparer == comparer._comparer);
-        }
 
-        public override int GetHashCode()
-        {
-            return _comparer.GetHashCode();
+            return _comparer?.Equals(comparer._comparer) ?? comparer._comparer == null;
         }
+
+        public override int GetHashCode() => _comparer?.GetHashCode() ?? 1592614637;
```

Comment:
Why not 0?  And if we do need to use something other than 0 and we go with that arbitrary value, it's more meaningful if written in hex.